### ISSUE TITLE
Error during build: "LLVM ERROR: Starting a frame before finishing the previous one!"

### ISSUE THREAD

**Issue by nalimilan** [JuliaLang/julia#15167]
I've seen this error for the first time today when building RPM nightlies. This is with system LLVM 3.7.1 + Julia patches and ORCJIT.
```
inference.jl
LLVM ERROR: Starting a frame before finishing the previous one!
Makefile:220: recipe for target '/builddir/build/BUILD/julia/build/usr/lib64/julia/inference.ji' failed
```
https://copr-be.cloud.fedoraproject.org/results/nalimilan/julia-nightlies/fedora-22-x86_64/00161544-julia/build.log.gz

**Comment by yuyichao** [comment]
Is this after increasing the stack limit and how much stack space are there?

**Comment by nalimilan** [comment]
Not sure what you mean by "after increasing the stack limit". I didn't change anything on the builders, but if a commit did that on master recently, then yes. Anyway, `ulimit` says `stack size  (kbytes, -s) 8192`.


I've just seen a different error when building julia-debug (on 32-bit this time). Instead of opening yet another issue, I'm posting the logs here, as it's likely related:
```
julia-debug: /builddir/build/BUILD/julia/src/toplevel.c:720: expr_to_lambda: Assertion `(((jl_value_t*)(((jl_taggedvalue_t*)((char*)(f) - sizeof(jl_taggedvalue_t)))->type_bits & ~(uintptr_t)15))==(jl_value_t*)(jl_expr_type)) && ((jl_expr_t*)f)->head == lambda_sym' failed.
/bin/sh: line 1: 12528 Aborted                 (core dumped) /builddir/build/BUILD/julia/build/usr/bin/julia-debug -C pentium4 --output-o /builddir/build/BUILD/julia/build/usr/lib/julia/sys-debug.o -f -J /builddir/build/BUILD/julia/build/usr/lib/julia/inference.ji sysimg.jl
```
https://copr-be.cloud.fedoraproject.org/results/nalimilan/julia-nightlies/fedora-23-i386/00161709-julia/build.log.gz

**Comment by yuyichao** [comment]
I meant https://github.com/JuliaLang/julia/issues/15079#issuecomment-184240329, since it is an stack overflow, there might be multiple ways it can cause errors.

**Comment by nalimilan** [comment]
Ah, sorry. Indeed, looks like `ulimit -s 100000` fixes it.

Though the error I reported in my last comment and the one from https://github.com/JuliaLang/julia/issues/15168 did not go away, so they are apparently unrelated to this one.

**Comment by vtjnash** [comment]
this will be fixed by https://github.com/JuliaLang/julia/pull/15300 very soon

**Comment by yuyichao** [comment]
This particular failure should be fixed by #15300 not. The stack usage is still very far from optimum (or what we had before). I've posted a few representative examples at https://github.com/JuliaLang/julia/issues/15369 and let's track the issue there.

**Comment by nalimilan** [comment]
Looks like it indeed works now without increasing the stack limit.
